tests/system/common.py

```diff
@@ -819,3 +822,26 @@ def kill_process_on_host(hostname, pattern):
     else:
         print("Killed no pids")
     return pids
+
+
+def service_available_predicate(service_name):
+    url = dcos_service_url(service_name)
+    try:
+        response = http.get(url)
+        return response.status_code == 200
+    except Exception as e:
+        return False
+
+
+# since exhibitor might take ~ 20 minutes to start up, we need to wait for it
+def wait_for_service_endpoint(service_name, timeout_sec=120):
+    """Checks the service url if available it returns true, on expiration
+    it returns false"""
+    @retrying.retry(wait_fixed=1000, stop_max_attempt_number=1200, retry_on_exception=ignore_exception)
+    def count_masters():
+        return len(get_all_masters())
+
+    master_count = count_masters()
+    return time_wait(lambda: service_available_predicate(service_name),
```

Comment:
If I am right, `time_wait` will return the number of seconds it took until the `predicate` function returned true. If there was a timeout, a `TimeoutExpired` exception will be thrown. I am not 100% sure if this is still correct on the shakedown version that we are using. Have a look..